packet_packer.go

```diff
@@ -20,6 +21,8 @@ type packedPacket struct {
 }
 
 type packetPacker struct {
+	mutex sync.Mutex
```

Comment:
I don't like it either. I think I forgot the mutex for `RetransmitNonForwardSecurePacket`. The mutex here protects two things: first the `controlFrames` slice in the `packetPacker`, and second the `packetNumberGenerator`.
Interestingly, the **only** place where a concurrent access might occur is when sending `ConnectionClose`s. So it would be nice if we could find a non-concurrent solution for that.